AzureMonitorAgent/agent.py

```diff
@@ -595,6 +598,26 @@ def enable():
                 output += "Output of '{0}':\n{1}".format(status_command, status_output)
                 return exit_code, output
 
+    # check if .NET is installed
+    check_dotnet, dotnetcmd_output = run_command_and_log("dotnet --list-runtimes",log_cmd=False)
+    if check_dotnet != 0:
+        print(".NET 7.0 is not installed. Please install .NET 7.0 if you are using Kql transformation. See more here https://learn.microsoft.com/en-us/dotnet/core/install/linux")
+        #ensure kql extension service is not running. do not block if it fails
+        kql_exit_code, disable_output = run_command_and_log(get_service_command("azuremonitor-kqlextension", "stop", "disable"))
+        if kql_exit_code != 0:
+            status_command = get_service_command("azuremonitor-kqlextension", "status")
+            kql_exit_code, status_output = run_command_and_log(status_command)
+    else:
+        if "7.0" in dotnetcmd_output:
+            print("Found .NET 7.0 installed.")
```

Comment:
need to start kql extension here

Reply:
Reminder to please start + enable. If we only start (but don't also enable), then the system will not start the service up after a reboot. We should typically be doing start/enable and stop/disable together.